virtual_ecosystem/models/plants/plants_model.py

```diff
@@ -790,11 +908,70 @@ def allocate_gpp(self) -> None:
                 )
             )
 
+            # FOURTH, ALLOCATE TO GROWTH:
+            # Grow the plants by increasing the stem dbh
+            # TODO: dimension mismatch (1d vs 2d array) - check in pyrealm
+            cohorts.dbh_values = cohorts.dbh_values + stem_allocation.delta_dbh
+
+            # Subtract the N/P required from growth from the element store, and
+            # redistribute it to the individual tisuses.
+            for stochiometry in stochiometries.values():
+                stochiometry.account_for_growth(stem_allocation)
+
+            # Balance the N & P surplus/deficit with the symbiote carbon supply
+            n_weighted_avg = np.dot(
+                self.data["ecto_supply_limit_n"][cell_id]
+                + self.data["arbuscular_supply_limit_n"][cell_id],
+                cohorts.n_individuals,
+            )
+            n_avaiable_per_cohort = n_weighted_avg / sum(n_weighted_avg)
+            n_available_per_stem = np.divide(
+                n_avaiable_per_cohort, cohorts.n_individuals
+            )
+            stochiometries["N"].element_surplus = (
+                stochiometries["N"].element_surplus + n_available_per_stem
+            )
+
+            p_weighted_avg = np.dot(
+                self.data["ecto_supply_limit_p"][cell_id]
+                + self.data["arbuscular_supply_limit_p"][cell_id],
+                cohorts.n_individuals,
+            )
+            p_available_per_cohort = p_weighted_avg / sum(p_weighted_avg)
+            p_available_per_stem = np.divide(
+                p_available_per_cohort, cohorts.n_individuals
+            )
+
+            stochiometries["P"].element_surplus = (
+                stochiometries["P"].element_surplus + p_available_per_stem
+            )
```

Comment:
Think we can loop this process?
```suggestion
            # Balance the N & P surplus/deficit with the symbiote carbon supply
            for element in ["N", "P"]:
                element_weighted_avg = np.dot(
                    self.data["ecto_supply_limit_" + element.lower()][cell_id]
                    + self.data["arbuscular_supply_limit_" + element.lower()][cell_id],
                    cohorts.n_individuals,
                )
                element_available_per_cohort = (
                    element_weighted_avg / sum(element_weighted_avg)
                )
                element_available_per_stem = np.divide(
                    element_available_per_cohort, cohorts.n_individuals
                )
                stochiometries[element].element_surplus += element_available_per_stem
```